src/index.js

```diff
@@ -11,53 +11,65 @@ class ApolloError extends ExtendableError {
   constructor (name, {
     message,
     time_thrown = (new Date()).toISOString(),
-    data = {}
+    data = {},
+    options = {},
   }) {
     const t = (arguments[2] && arguments[2].thrown_at) || time_thrown;
     const d = Object.assign({}, data, ((arguments[2] && arguments[2].data) || {}));
+    const opts = Object.assign({}, options, ((arguments[2] && arguments[2].options) || {}));
 
     super(serializeName([
       name,
       t,
       Object.assign({}, d, {
         toString: () => JSON.stringify(d)
-      })
+      }),
     ]));
 
     this._name = name;
     this._humanized_message = message || '';
     this._time_thrown = t;
     this._data = d;
+    this._locations = (opts.showLocations && arguments[2] && arguments[2].locations)
```

Comment:
It's a truthy check of the opts. The line was initially below `const t = (arguments[2] && arguments[2]...` and I was keeping it brief, but I'll make the code a little clearer tomorrow!

The arguments come from the originalError in formatError.